### PR TITLE
vmount: refactor to auto-select between all inputs

### PR DESCRIPTION
**Pull request by julianoes**
It is not convenient having to change a parameter to change a gimbal
from RC input to mavlink input mode or back. This refactor changes the
behaviour to use whatever is available, RC or mavlink commands.

Once a mavlink command is followed, control can be taken back using RC,
however, this requires a clear stick change.

Needs reviewing and testing.

As discussed with @bkueng.

### CODE REVIEW

**Comment by LorenzMeier** [comment]
Cool! Looking forward to test this!

File: src/drivers/vmount/input_mavlink.cpp [hunk 1]
```diff
@@ -121,38 +110,34 @@ int InputMavlinkROI::update_impl(unsigned int timeout_ms, ControlData **control_
...
 			orb_copy(ORB_ID(vehicle_roi), _vehicle_roi_sub, &vehicle_roi);
 
-			if (vehicle_roi.mode == vehicle_roi_s::VEHICLE_ROI_NONE) {
-				_allow_manual_control = true;
+			_control_data.gimbal_shutter_retract = false;
 
-				if (!_manual_control) {
-					_control_data.type = ControlData::Type::Neutral;
-				}
+			if (vehicle_roi.mode == vehicle_roi_s::VEHICLE_ROI_NONE) {
```
Comment (julianoes) [inline src/drivers/vmount/input_mavlink.cpp]: But we are not setting `control_data` in here anyway, so it will be `NULL`.

Comment (julianoes) [inline src/drivers/vmount/input_mavlink.cpp]: Ok or you set != NULL and Neutral, got it.

**Comment by julianoes** [comment]
Flight tested, works.

**Comment by mhkabir** [comment]
We need docs for this :)

**Pull request merged**